pkg/asset/machines/openstack/machines.go

```diff
@@ -78,36 +78,57 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, az string, role, userDataSecret string, trunk string) (*openstackprovider.OpenstackProviderSpec, error) {
+func provider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, az string, role, userDataSecret string, trunk string, additionalSubnets []string, additionalSecurityGroups []string) (*openstackprovider.OpenstackProviderSpec, error) {
```

Comment:
We will have to add a separate arg for additional subnets for BYO network, so it would be better to name the additional worker subnets "additionalWorkerSubnets"

```suggestion
func provider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, az string, role, userDataSecret string, trunk string, additionalWorkerSubnets []string, additionalSecurityGroups []string) (*openstackprovider.OpenstackProviderSpec, error) {
```